# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1264029401 -3600
#      Thu Jan 21 00:16:41 2010 +0100
# Node ID f481496013a265952008f21162c150ce1bb51949
# Parent  d0ce5e973937be7f52e4100909f2794031630fc3
Set default line style in __contour__

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,19 +1,23 @@
+2010-01-20  David Bateman  <dbateman@free.fr>
+
+	* plot/private/__contour__.m: Set default linestyle if pltopt returns
+	an empty style.
+
 2010-01-20  John W. Eaton  <jwe@octave.org>
 
 	* time/datetick.m: Make form arg optional.
 	From Philip Nienhuis <pr.nienhuis@hccnet.nl>.
 
 2010-01-20  David Bateman  <dbateman@free.fr>
 
 	* plot/__go_draw_axes__.m: Correct an error in drawing of patch
 	objects where unrequested markers were drawn.
 
-
 	* plot/__go_draw_figure__.m: Also plot white foreground and
 	background colors.
 	* plot/plotyy.m: Set axes colors to "none".
 
 	* plot/__go_draw_axes__.m (next_marker):  Remove and all uses of
 	this function.
 	Partially respect the markeredgecolor and markerfacecolor properties.
 	* plot/private/__color__str_rgb__.m: New function
diff --git a/scripts/plot/private/__contour__.m b/scripts/plot/private/__contour__.m
--- a/scripts/plot/private/__contour__.m
+++ b/scripts/plot/private/__contour__.m
@@ -31,16 +31,19 @@ function [c, hg] = __contour__ (varargin
   edgecolor = "flat";
   for i = 3 : nargin
     arg = varargin {i};
     if ((ischar (arg) || iscell (arg)))
       [linespec, valid] = __pltopt__ ("__contour__", arg, false);
       if (isempty (linespec.color))
 	linespec.color = "auto";
       endif
+      if (isempty (linespec.linestyle))
+	linespec.linestyle = "-";
+      endif
       if (valid)
 	have_line_spec = true;
 	varargin(i) = [];
 	break;
       endif
     endif
   endfor
 
